README.md

```diff
@@ -91,3 +91,41 @@ For example, to set the minimum log level using the _Windows_ command prompt:
 set Serilog:MinimumLevel=Debug
 dotnet run
 ```
+
+### Nested configuration sections
+
+Some Serilog packages require a reference to a logger configuration object. These must be "hinted" by appending a `>` character to the end of the argument name. The sample program in this project illustrates this with the following entry configuring the _Serilog.Sinks.Async_ package to wrap the _Serilog.Sinks.File_ package.
```

Comment:
This seems like a bit of a UX regression, and it's not immediately clear when this will or won't be a breaking change; can't we work out what kind of argument we're dealing with by reflecting over the parameters? I am afraid I'm missing something here, sorry - struggling to keep up :-)

Reply:
I'll take another look, but other than the Async wrapper package, frankly I'm not clear on the use cases for nested config. (For example, the reason for the logger nested config in the sample code and tests makes no sense to me.) I don't really like the hack myself, but at the point the code is "deciding" how to store the config section (the parameter value, effectively), we don't yet know what the parameter type is. "How to store" means the various parameter-value classes (string, config, or my new object class). I suspect the config value class can go away and that can be added to the object value class.

Reply:
Fixed. Very glad you questioned this -- the "hint" hack is gone and the code is simplified! I took another look at what was going on in `ConfigurationSectionArgumentValue` and it turns out I could get rid of that and move the functionality into `ObjectArgumentValue`. The trick was deferring the new `ConfigurationReader` pass to read the nested config until we are actually calling the target method. Now the read is triggered by the `Action<>` parameter's type rather than "guessing" based on config structure.

This made me notice an edge-case risk of error based on how this package uses the configuration section `Value` property that I will open an issue about just to document it (nothing relating to my changes). It's extremely unlikely to occur in the wild but maybe somebody will feel like addressing it.